amy/fiscal/tests/test_membership.py

```diff
@@ -930,6 +991,65 @@ def test_mix_adding_removing_members_from_consortium(self):
 
         self.assertEqual(list(self.membership.organizations.all()), [self.org_beta])
 
+    def test_editing_noneditable_members_fails(self):
+        """Ensure an attempt to edit member without 'editable' checkbox ticked off
+        fails with validation error."""
+        self.setUpMembership(consortium=True)
+        m1 = Member.objects.create(
+            organization=self.org_alpha,
+            membership=self.membership,
+            role=self.member_role,
+        )
+
+        data = {
+            "form-TOTAL_FORMS": 1,
+            "form-INITIAL_FORMS": 1,
+            "form-MIN_NUM_FORMS": 0,
+            "form-MAX_NUM_FORMS": 1000,
+            "form-0-organization": self.org_beta.pk,
+            "form-0-role": m1.role.pk,
+            "form-0-id": m1.pk,
+        }
+        response = self.client.post(
+            reverse("membership_members", args=[self.membership.pk]),
+            data=data,
+            follow=True,
+        )
+
+        self.assertEqual(response.status_code, 200)  # form failed
+        self.assertIn("__all__", response.context["formset"].errors[0])
```

Comment:
You may want to assert the specific error about form values not changing is in the __all__ list.

Reply:
Resolved by checking for `response.context["formset"].errors[0]` complete value.